front/ui/storybook/stories/ui-core/ComboBoxCustomSuggestions.stories.tsx

```diff
@@ -0,0 +1,454 @@
+        const namePrefix = nameNorm.startsWith(tokenNorm);
+        const nameContains = matchesName && !namePrefix;
+        const chPrefix = suggestion.chList.some((ch) =>
+          ch.code.toUpperCase().startsWith(tokenUpper)
+        );
+        const chContains = matchesCh && !chPrefix;
+
+        let category = 6;
+        if (trigramExact) category = 0;
+        else if (trigramPrefix) category = 1;
+        else if (namePrefix) category = 2;
+        else if (nameContains) category = 3;
+        else if (chPrefix) category = 4;
+        else if (chContains) category = 5;
+
+        return { suggestion, category };
+      })
+      .filter((entry) => entry !== null);
+
+    const ordered = scored
+      .sort(
+        (a, b) =>
+          a.category - b.category || a.suggestion.name.localeCompare(b.suggestion.name, 'fr')
+      )
+      .map((entry) => entry.suggestion);
+
+    return ordered.map((suggestion) => ({
+      ...suggestion,
+      chList: suggestion.chList.map((ch) => ({
+        ...ch,
+        isCandidate: true,
+        isBestSuggestion: false,
+      })),
+    }));
+  }
+
+  // CASE 2 : several words
+  const baseText = tokens.slice(0, -1).join(' ');
+  const lastToken = tokens[tokens.length - 1];
+
+  const baseNorm = normalize(baseText);
+  const baseUpper = baseText.toUpperCase();
+  const lastTokenUpper = lastToken.toUpperCase();
+
+  const exactTrigramAndChMatches = allSuggestions
+    .filter(
+      (s) =>
+        s.trigram.toUpperCase() === baseUpper &&
+        s.chList.some((ch) => ch.code.toUpperCase() === lastTokenUpper)
+    )
+    .map((s) => ({
+      ...s,
+      chList: s.chList.map((ch) => {
+        const isExactCh = ch.code.toUpperCase() === lastTokenUpper;
+        return {
+          ...ch,
+          isCandidate: isExactCh,
+          isBestSuggestion: isExactCh,
+        };
+      }),
+    }));
+
+  if (exactTrigramAndChMatches.length > 0) {
+    return exactTrigramAndChMatches;
+  }
+
+  const scoredPreliminary = allSuggestions
+    .map((suggestion) => {
+      const nameNorm = normalize(suggestion.name);
+      const trigramUpper = suggestion.trigram.toUpperCase();
+
+      const chPrefixMatch = suggestion.chList.some((ch) =>
+        ch.code.toUpperCase().startsWith(lastTokenUpper)
+      );
+
+      // If last token doesn't start like a known CH, we keep the full text matching
+      if (!chPrefixMatch) {
+        const fullNorm = normalize(query);
+        const fullUpper = query.toUpperCase();
+
+        const matchesFullName = nameNorm.includes(fullNorm);
+        const matchesFullTrig = trigramUpper.includes(fullUpper);
+
+        if (!matchesFullName && !matchesFullTrig) return null;
+
+        return {
+          suggestion: {
+            ...suggestion,
+            chList: suggestion.chList.map((ch) => ({
+              ...ch,
+              isCandidate: true,
+              isBestSuggestion: false,
+            })),
+          },
+          baseScore: 99,
```

Comment:
If you handle also the case where only 1 word has been entered here, you will have to use your new function `getBaseScore`